api/app/tests/apps/email_integrations/reporter/test_email_templates.py

```diff
@@ -10,25 +10,37 @@
 
 class TestEmailTemplates(TestCase):
     def setUp(self):
-        def get_email():
-            return f'{uuid.uuid4()}@example.com'
-
         self.email_template = EmailTemplate.objects.create(
             key=EmailTemplate.SIGNAL_CREATED,
             title='Template title {{ signal.id }}',
-            body='# Template title\n Thanks a lot for reporting **{{ signal.id }}**',
+            body='# Template title\n Thanks a lot for reporting **{{ signal.id }}** {{ signal.text }}',
         )
 
-        self.signal = SignalFactory.create(reporter__email=get_email())
-
     def test_email_template(self):
+        signal = SignalFactory.create(reporter__email=self.get_email())
+
         ma = MailActions(mail_rules=SIGNAL_MAIL_RULES)
-        ma.apply(signal_id=self.signal.id)
+        ma.apply(signal_id=signal.id)
 
-        self.assertEqual(f'Template title {self.signal.id}', mail.outbox[0].subject)
-        self.assertEqual(f'Template title\n\nThanks a lot for reporting {self.signal.id}\n\n', mail.outbox[0].body)
+        self.assertEqual(f'Template title {signal.id}', mail.outbox[0].subject)
+        self.assertEqual(f'Template title\n\nThanks a lot for reporting {signal.id} {signal.text}\n\n',
+                         mail.outbox[0].body)
 
         body, mime_type = mail.outbox[0].alternatives[0]
         self.assertEqual(mime_type, 'text/html')
         self.assertIn('<h1>Template title</h1>', body)
-        self.assertIn(f'<p>Thanks a lot for reporting <strong>{self.signal.id}</strong></p>', body)
+        self.assertIn(f'<p>Thanks a lot for reporting <strong>{signal.id}</strong>', body)
+
+    def test_evil_input(self):
+        evil_signal = SignalFactory.create(reporter__email=self.get_email(),
+                                           text='<script>alert("something evil");</script>')
+
+        ma = MailActions(mail_rules=SIGNAL_MAIL_RULES)
+        ma.apply(signal_id=evil_signal.id)
+
+        self.assertEqual(f'Template title {evil_signal.id}', mail.outbox[0].subject)
+        self.assertEqual(f'Template title\n\nThanks a lot for reporting {evil_signal.id} alert("something evil");\n\n',
+                         mail.outbox[0].body)
+
```

Comment:
Not a blocker but: I would like a test added that demonstrates the 's-Hertogenbosch problem (and that it is solved).

Reply:
Added an explicit test for this problem 👍

Reply:
thanks!